plugin/evm/limit_order.go

```diff
@@ -120,6 +122,7 @@ func (lop *limitOrderProcesser) ListenAndProcessTransactions() {
 }
 
 func (lop *limitOrderProcesser) RunBuildBlockPipeline() {
+	defer recoverPanic("runBuildBlockPipeline failure", "build_block_pipeline_panics")
```

Comment:
how about creating a Recover wrapper which accepts a function and automatically recovers in case of panic? And it can automatically fetch the function name from the function itself using reflection, and use that to log messages and update metrics.

It can be used like this:
https://github.com/hubble-exchange/subnet-evm/blob/48de40420cb9ad69cce723d88ce3bbedcea3efaf/plugin/evm/block_builder.go#L167